### PR TITLE
adding templates for issues

### PR DESCRIPTION
**Pull request by GFMoraga**
# Change Summary

## Overview
Adding template for issue. Updates on algorithim_dev.md, bug_reporting.md

## New Files
General_issue.md

### CODE REVIEW

File: .github/issue_templates/Algorithim_development.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+```python
+# Code
+```
+
+### Requirements/ Dependencies:
+
+- [ ] 
+    1. Yes
+- [ ] 
+    2. No
```
Comment (greglucas) [inline .github/issue_templates/Algorithim_development.md]: This section doesn't render quite right in Markdown: 
image

Using checkboxes will show "1/2" items completed, so I don't think we want them. Perhaps just remove these and put `<!-- Specify if any dependencies were added -->` as a comment line in here?

File: .github/issue_templates/Bug_reporting.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+    Where did it occur?
```
Comment (greglucas) [inline .github/issue_templates/Bug_reporting.md]:
```suggestion
## Description of the issue
```
I'm not sure what you mean by "where" here. Perhaps start this out with letting the user just put in a brief description and then follow with your more precise requests below.

File: .github/issue_templates/Bug_reporting.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+### Actual behavior (What does happen)
+
+### Additional notes
+
+### Did you try to fix the issue? (Yes/No)
+
+- [ ] Yes
+- [ ] No
+- [ ] I tried but I couldn't fix it
```
Comment (greglucas) [inline .github/issue_templates/Bug_reporting.md]:
```suggestion
### Suggested fix?
```
Maybe leave a spot for people to add a suggested fix? But, I don't think we need to know if they tried or not, they can open a PR if they have a fix for it.

**Review by greglucas — COMMENTED** [review]
I think this content is good and the PR looks like it is pointing at the right locations!

I think we might want to use `yaml` formatting instead to allow for some more flexibility in how it is rendered with the boxes/text areas?

https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms

Can you take the content/ideas you have here and convert them over into:
`algorithm_development.yml` and `bug_report.yml`

I think this would help with some of your multi-section pieces here as well and allow drop-downs, optional sections etc...

File: .github/issue_templates/Bug_reporting.md (hunk 3 above)
Comment (tech3371) [reply .github/issue_templates/Bug_reporting.md]: I agree. We could leave last one out.

File: .github/issue_templates/Algorithim_development.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+### Algorithm Description:
+
+### Requirements:
+
+> **Specify: (optional)**
+
+### Algorithm Code Info: Data/Input/Output
+
```
Comment (tech3371) [inline .github/issue_templates/Algorithim_development.md:16]: We could even break down this section with subsections. Example, we could have subsection such as input data description, algorithm steps or pseudo code, output data descriptions.

File: .github/issue_templates/Algorithim_development.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+### Requirements/ Dependencies:
+
+- [ ] 
+    1. Yes
+- [ ] 
+    2. No
+
+### If yes, specify:
+
+### Other Notes:
```
Comment (maxinelasp) [inline .github/issue_templates/Algorithim_development.md:28]: I'd like to see a section for related issues as well, to encourage people to link them together.

File: .github/issue_templates/Bug_reporting.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+```python
+# Code
+```
+
+### Expected behavior (What should happen)
+
+### Actual behavior (What does happen)
+
+### Additional notes
```
Comment (maxinelasp) [inline .github/issue_templates/Bug_reporting.md]: Maybe a section for affected code, processes, or data might be a helpful addition? 
```suggestion
### Additional notes

### Affected areas (code, data, or processes)
```

**Review by maxinelasp — CHANGES_REQUESTED** [review]

**Comment by GFMoraga** [comment]
@greglucas what are the benefits of a .yml and is it as easy as changing the .md to .yml?

**Comment by GFMoraga** [comment]
@greglucas @tech3371 why does this not specify the whitespace? Can you help me understand this block?

trim trailing whitespace.................................................Failed
- hook id: trailing-whitespace
- exit code: 1
- files were modified by this hook

**Comment by greglucas** [comment]
> @greglucas what are the benefits of a .yml and is it as easy as changing the .md to .yml?

I think that is what it takes to get the nice auto-formatting shown in that link with text-boxes etc. (i.e. GitHub has special handling of these files) But, maybe that can be done with Markdown as well? I don't know, I've just only seen other repos use yml. This is fine for a first cut and if we do want that format it can be a follow-on ticket as well.

> @greglucas @tech3371 why does this not specify the whitespace? Can you help me understand this block?

All files should have a blank line at the end of them. You can also add a comment to your PR to have the bot help you out as well I think, with just this text in a new comment: `pre-commit.ci autofix`

**Comment by GFMoraga** [comment]
pre-commit.ci autofix

**Review by bourque — APPROVED** [review]

**Comment by GFMoraga** [comment]
Markdown is two things: (1) a plain text formatting syntax; and (2) a software tool, written in Perl, that converts the plain text formatting to HTML. On the other hand, YAML is detailed as "A straightforward machine parsable data serialization format designed for human readability and interaction". A human-readable data-serialization language. It is commonly used for configuration files but could be used in many applications where data is being stored or transmitted.

**Review by maxinelasp — APPROVED** [review]

**Pull request merged**